src/main/java/com/aws/greengrass/secretmanager/SecretManagerService.java

```diff
@@ -216,14 +209,15 @@
 
     /**
      * Handles secret API calls from lambda-manager for v1 style lambdas.
-     * @param serviceName  Lambda component name requesting the secret.
-     * @param request      v1 style get secret request
+     *
+     * @param serviceName Lambda component name requesting the secret.
+     * @param request     v1 style get secret request
      * @return v1 style secret or error serialized as JSON bytes. If a serialization error occurs, null is returned.
      */
+    // Note: Do not rename/move this method around as it is used via reflection from the lambda-manager.
     public byte[] getSecret(String serviceName, byte[] request) {
-        logger.atInfo().event("secret-access")
-                .kv("Principal", serviceName)
-                .kv("secret", new String(request))
+
+        logger.atInfo().event("secret-access").kv("Principal", serviceName).kv("secret", new String(request))
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

#### new String [](|8df4fb67-a24a-4a26-a18a-395ef71250d0|)
Java Strings use Unicode internally to encode characters. 

When constructing a string from a `byte[]`, a character encoding must be used to convert the bytes to a String&#39;s internal encoding. When no character encoding is passed explicitly, the Java runtime will use a default encoding. This encoding usually isn&#39;t UTF-8 and might become misconfigured. When converting bytes to characters or strings, always supply the actual character encoding to use.

#### Suggested solutions:
Specify an encoding

*(likely `UTF-8`, which is backward compatible with ASCII and has multi-language support)*

```
String message = new String(messageBytes, StandardCharsets.UTF_8);
```